web/src/compose.js

```diff
@@ -363,44 +364,74 @@ export function render_and_show_preview($preview_spinner, $preview_content_box,
         } else {
             rendered_preview_html = rendered_content;
         }
-
-        $preview_content_box.html(util.clean_user_content_links(rendered_preview_html));
+        if (!Number.isNaN(message_number)) {
+            // a valid number means that the message should be enumerated
+            const params = {
+                message_number,
+                rendered_preview_html,
+            };
+            const enumerated_rendered_preview_html = render_enumerated_split_message_part(params);
+            $preview_content_box.append(
+                $(util.clean_user_content_links(enumerated_rendered_preview_html)),
+            );
+        } else {
+            $preview_content_box.html(util.clean_user_content_links(rendered_preview_html));
+        }
         rendered_markdown.update_elements($preview_content_box);
     }
 
     if (content.length === 0) {
-        show_preview($t_html({defaultMessage: "Nothing to preview"}));
+        show_preview($t_html({defaultMessage: "Nothing to preview"}), Number.NaN);
```

Comment:
Any reason for passing Nan instead of skipping this argument? This seems overcomplicated

Reply:
I was just trying to make things more explicit. I'll remove it.